yt/frontends/ytdata/data_structures.py

```diff
@@ -732,19 +732,24 @@ def _is_valid(self, *args, **kwargs):
 
 class YTProfileDataset(YTNonspatialDataset):
     """Dataset for saved profile objects."""
+    fluid_types = ("data", "gas", "standard_deviation")
     def __init__(self, filename, unit_system="cgs"):
         super(YTProfileDataset, self).__init__(filename,
                                                unit_system=unit_system)
 
+    _profile = None
     @property
     def profile(self):
-        if self.dimensionality == 1:
-            return Profile1DFromDataset(self)
-        if self.dimensionality == 2:
-            return Profile2DFromDataset(self)
-        if self.dimensionality == 3:
-            return Profile3DFromDataset(self)
-        return None
+        if self._profile is None:
```

Comment:
Any chance you could switch these, so that if `self._profile is not None` it returns early, and then this section of the conditional lives one level of indentation higher?

Reply:
Sure, I can do that. I did it this way as having the single return underneath the creation step seemed like standard procedure for properties.